src/sentry/api/serializers/models/alert_rule_trigger_action.py

```diff
@@ -39,7 +42,45 @@ def get_input_channel_id(self, action):
         """
         return action.target_identifier if action.type == action.Type.SLACK.value else None
 
-    def serialize(self, obj, attrs, user):
+    def get_attrs(self, item_list, user, **kwargs):
+        sentry_app_ids = {action.sentry_app_id for action in item_list}
+        sentry_app_installations = SentryAppInstallation.objects.get_installed_for_organization(
+            kwargs["organization_id"]
+        ).filter(sentry_app_id__in=sentry_app_ids)
+        sentry_app_installations_by_sentry_app_id = {
+            installation.sentry_app_id: installation for installation in sentry_app_installations
+        }
+
+        components = SentryAppComponent.objects.filter(
+            sentry_app_id__in=sentry_app_ids, type="alert-rule-action"
+        )
```

Comment:
Could these queries be combined using eager loading?

Reply:
@markstory ended up refactoring to move out all the SentryApp lookups to the endpoint and pass down as kwargs. I combined the queries using `.annotate` to get the related sentry_app_component for an installation.